src/tasks/minions/farmingActivity.ts

```diff
@@ -0,0 +1,426 @@
+
+		// check bank for farmer's items
+		if (bankHasItem(userBank, itemID(`Farmer's strawhat`))) {
+			bonusXpMultiplier += 0.004;
+			farmersPiecesCheck++;
+		}
+		if (
+			bankHasItem(userBank, itemID(`Farmer's jacket`)) ||
+			bankHasItem(userBank, itemID(`Farmer's shirt`))
+		) {
+			bonusXpMultiplier += 0.008;
+			farmersPiecesCheck++;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boro trousers`))) {
+			bonusXpMultiplier += 0.006;
+			farmersPiecesCheck++;
+		}
+		if (bankHasItem(userBank, itemID(`Farmer's boots`))) {
+			bonusXpMultiplier += 0.002;
+			farmersPiecesCheck++;
+		}
+		if (farmersPiecesCheck === 4) bonusXpMultiplier += 0.005;
+
+		let loot = {
+			[itemID('Weeds')]: 0
+		};
+
+		delete loot[itemID('Weeds')];
+
+		if (!patchType.patchPlanted) {
+			if (!plant) return;
+
+			rakeXp = quantity * 4 * 3; // # of patches * exp per weed * # of weeds
+			plantXp = quantity * (plant.plantXp + compostXp);
+			farmingXpReceived = plantXp + harvestXp + rakeXp;
+
+			loot[itemID('Weeds')] = quantity * 3;
+
+			let str = `${user}, ${
+				user.minionName
+			} finished raking ${quantity} patches and planting ${quantity}x ${
+				plant.name
+			}.\nYou received ${plantXp.toLocaleString()} XP from planting and ${rakeXp.toLocaleString()} XP from raking for a total of ${farmingXpReceived.toLocaleString()} Farming XP.`;
+
+			bonusXP += Math.floor(farmingXpReceived * bonusXpMultiplier);
+			if (bonusXP > 0) {
+				str += ` You received an additional ${bonusXP.toLocaleString()} in bonus XP.`;
+			}
+
+			await user.addXP(SkillsEnum.Farming, Math.floor(farmingXpReceived + bonusXP));
+			const newLevel = user.skillLevel(SkillsEnum.Farming);
+
+			if (newLevel > currentFarmingLevel) {
+				str += `\n\n${user.minionName}'s Farming level is now ${newLevel}!`;
+			}
+
+			if (Object.keys(loot).length > 0) {
+				str += `\n\nYou received: ${await createReadableItemListFromBank(
+					this.client,
+					loot
+				)}.`;
+			}
+
+			await user.addItemsToBank(loot, true);
+
+			const updatePatches: PatchTypes.PatchData = {
+				lastPlanted: plant.name,
+				patchPlanted: true,
+				plantTime: currentDate + duration,
+				lastQuantity: quantity,
+				lastUpgradeType: upgradeType,
+				lastPayment: patchType.lastPayment
+			};
+
+			await user.settings.update(getPatchType, updatePatches);
+
+			str += `\n\n${user.minionName} tells you to come back after your plants have finished growing!`;
+
+			const channel = this.client.channels.get(channelID);
+			if (!channelIsSendable(channel)) return;
+
+			channel.send(str);
+		} else if (patchType.patchPlanted) {
+			const plantToHarvest = Farming.Plants.find(
+				plant => plant.name === patchType.lastPlanted
+			);
+			if (!plantToHarvest) return;
+			if (!plant) return;
+
+			let quantityDead = 0;
+			for (let i = 0; i < patchType.lastQuantity; i++) {
+				for (let j = 0; j < plantToHarvest.numOfStages - 1; j++) {
+					const checkIfDied = Math.random();
```

Comment:
this name doesnt make sense to me. `checkIfDied` is a number that is uses TO check if it died, what about `deathRoll` or something like that.